tests/integration_test/src/action_handlers_flare_api.py

```diff
@@ -0,0 +1,175 @@
+            if len(command_args) == 2:
+                # if server id is provided
+                server_ids = [command_args[1]]
+            else:
+                # start all servers
+                server_ids = list(admin_controller.site_launcher.server_properties.keys())
+            for sid in server_ids:
+                admin_controller.site_launcher.start_server(sid)
+        elif command_args[0] == "client":
+            if len(command_args) == 2:
+                # if client id is provided
+                client_ids = [command_args[1]]
+            else:
+                # start all clients
+                client_ids = list(admin_controller.site_launcher.client_properties.keys())
+            for cid in client_ids:
+                admin_controller.site_launcher.start_client(cid)
+        elif command_args[0] == "overseer":
+            admin_controller.site_launcher.start_overseer()
+        else:
+            raise RuntimeError(f"Target {command_args[0]} is not supported.")
+
+
+class _KillHandler(_CmdHandler):
+    def handle(self, command_args: list, admin_controller: NVFFLAREAPITestDriver, admin_api: Session):
+        if command_args[0] == "server":
+            if len(command_args) == 2:
+                # if server id is provided
+                server_id = command_args[1]
+            else:
+                # kill active server
+                server_id = admin_controller.site_launcher.get_active_server_id(admin_api.api.port)
+            admin_controller.site_launcher.stop_server(server_id)
+        elif command_args[0] == "overseer":
+            admin_controller.site_launcher.stop_overseer()
+        elif command_args[0] == "client":
+            if len(command_args) == 2:
+                # if client id is provided
+                client_ids = [command_args[1]]
+            else:
+                # close all clients
+                client_ids = list(admin_controller.site_launcher.client_properties.keys())
+            for cid in client_ids:
+                admin_api.api.do_command("remove_client " + admin_controller.site_launcher.client_properties[cid].name)
+                admin_controller.site_launcher.stop_client(cid)
+
+
+class _SleepHandler(_CmdHandler):
+    def handle(self, command_args: list, admin_controller: NVFFLAREAPITestDriver, admin_api: Session):
+        time.sleep(int(command_args[0]))
+
+
+class _AdminCommandsHandler(_CmdHandler):
+    def handle(self, command_args: list, admin_controller: NVFFLAREAPITestDriver, admin_api: Session):
+        pass
+        # run_admin_api_tests(admin_api)
+
+
+class _NoopHandler(_CmdHandler):
+    def handle(self, command_args: list, admin_controller: NVFFLAREAPITestDriver, admin_api: Session):
+        pass
+
+
+class _TestDoneHandler(_CmdHandler):
+    def handle(self, command_args: list, admin_controller: NVFFLAREAPITestDriver, admin_api: Session):
+        admin_controller.test_done = True
+
+
+class _SubmitJobHandler(_CmdHandler):
```

Comment:
Probably it's better to build a AdminCommandHandler which can run different kinds of admin commands, instead of build a Handler for each individual admin command.